go/api/config.go

```diff
@@ -53,87 +77,117 @@ func (config *baseClientConfiguration) toProtobufConnRequest() *protobuf.Connect
 	}
 
 	if config.credentials != nil {
+		if err := config.credentials.validate(); err != nil {
+			return nil, err
+		}
+
 		authInfo := protobuf.AuthenticationInfo{}
-		if config.credentials.Username != "" {
-			authInfo.Username = config.credentials.Username
+		if config.credentials.Username != nil {
+			authInfo.Username = *config.credentials.Username
 		}
-		authInfo.Password = config.credentials.Password
+		authInfo.Password = *config.credentials.Password
 		request.AuthenticationInfo = &authInfo
 	}
 
 	request.ReadFrom = mapReadFrom(config.readFrom)
-	if config.requestTimeout > 0 {
-		request.RequestTimeout = uint32(config.requestTimeout)
+	if config.requestTimeout != nil {
+		request.RequestTimeout = *config.requestTimeout
 	}
 
-	return &request
+	return &request, nil
 }
 
 func mapReadFrom(readFrom ReadFrom) protobuf.ReadFrom {
-	if readFrom == PREFER_REPLICA {
+	if readFrom == PreferReplica {
 		return protobuf.ReadFrom_PreferReplica
 	}
 
 	return protobuf.ReadFrom_Primary
 }
 
-// BackoffStrategy represents the strategy used to determine how and when to reconnect, in case of
-// connection failures. The time between attempts grows exponentially, to the formula
-// rand(0 ... factor * (exponentBase ^ N)), where N is the number of failed attempts.
+// BackoffStrategy represents the strategy used to determine how and when to reconnect, in case of connection failures. The
+// time between attempts grows exponentially, to the formula rand(0 ... factor * (exponentBase ^ N)), where N is the number of
```

Comment:
are there any code highlighting/emphasizing features in go code docs?

Reply:
Go docs support paragraphs, headings, links, lists, and preformatted code blocks. They do not seem to have bold, italics, or inline code highlighting

Reply:
Please try using them, for example for this formula
https://github.com/Bit-Quill/glide-for-redis/blob/e71c8e66e0ec7ac7ea89dfc95ce584277a415e0e/java/client/src/main/java/glide/api/models/configuration/BackoffStrategy.java#L10-L11